cleanrl/ppo_memory_env.py

```diff
@@ -0,0 +1,369 @@
+        help='if toggled, this experiment will be tracked with Weights and Biases')
+    parser.add_argument('--wandb-project-name', type=str, default="cleanRL",
+        help="the wandb's project name")
+    parser.add_argument('--wandb-entity', type=str, default=None,
+        help="the entity (team) of wandb's project")
+    parser.add_argument('--capture-video', type=lambda x:bool(strtobool(x)), default=False, nargs='?', const=True,
+        help='weather to capture videos of the agent performances (check out `videos` folder)')
+
+    # Algorithm specific arguments
+    parser.add_argument('--num-envs', type=int, default=1,
+        help='the number of parallel game environments')
+    parser.add_argument('--num-steps', type=int, default=128,
+        help='the number of steps to run in each environment per policy rollout')
+    parser.add_argument('--anneal-lr', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+        help="Toggle learning rate annealing for policy and value networks")
+    parser.add_argument('--gae', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+        help='Use GAE for advantage computation')
+    parser.add_argument('--gamma', type=float, default=0.99,
+        help='the discount factor gamma')
+    parser.add_argument('--gae-lambda', type=float, default=0.95,
+        help='the lambda for the general advantage estimation')
+    parser.add_argument('--num-minibatches', type=int, default=1,
+        help='the number of mini-batches')
+    parser.add_argument('--update-epochs', type=int, default=10,
+        help="the K epochs to update the policy")
+    parser.add_argument('--norm-adv', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+        help="Toggles advantages normalization")
+    parser.add_argument('--clip-coef', type=float, default=0.2,
+        help="the surrogate clipping coefficient")
+    parser.add_argument('--clip-vloss', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+        help='Toggles wheter or not to use a clipped loss for the value function, as per the paper.')
+    parser.add_argument('--ent-coef', type=float, default=0.01,
+        help="coefficient of the entropy")
+    parser.add_argument('--vf-coef', type=float, default=0.5,
+        help="coefficient of the value function")
+    parser.add_argument('--max-grad-norm', type=float, default=0.5,
+        help='the maximum norm for the gradient clipping')
+    parser.add_argument('--target-kl', type=float, default=None,
+        help='the target KL divergence threshold')
+    args = parser.parse_args()
+    args.batch_size = int(args.num_envs * args.num_steps)
+    args.minibatch_size = int(args.batch_size // args.num_minibatches)
+    # fmt: on
+    return args
+
+
+class TestMemoryEnv(gym.Env):
+    """
+    final state  [0,0,0,1,0]    [0,0,0,0,1]
```

Comment:
First, why is the cartpole state length 5 instead of 4?
Second, why is it only integers and 5 states?
Is there any reference of this env?

Reply:
I made up this env.